src/App.tsx

```diff
@@ -1,34 +1,133 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+enum FilterBy {
+  All = 'all',
+  COMPLETED = 'completed',
+  ACTIVE = 'active',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [query, setQuery] = useState<string>('');
```

Comment:
it's extra, TS already knows about this types, bc you put an initial value
```suggestion
  const [isLoading, setIsLoading] = useState(false);
  const [query, setQuery] = useState('');
```